runhouse/resources/hardware/cluster.py

```diff
@@ -1643,9 +1648,8 @@ def run(
         from runhouse.resources.envs import Env
 
         if self.on_this_cluster() and node:
-            raise ValueError(
-                "Cannot specify a node when running from within the cluster."
-            )
+            # Switch the external ip to an internal ip
+            node = self.internal_ips[self.ips.index(node)]
```

Comment:
confused here - both ips refer to internal ips and doesn't this just return the same node?